src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -14,21 +14,23 @@ public interface IConfigurationProvider : IExtension, IAgentService
     {
         string ConfigurationProviderType { get; }
 
-        string GetServerUrl(CommandSettings command);
+        void GetServerUrl(AgentSettings agentSettings, CommandSettings command);
 
-        Task TestConnectionAsync(string tfsUrl, VssCredentials creds);
+        Task TestConnectionAsync(AgentSettings agentSettings, VssCredentials creds);
 
-        Task<int> GetPoolId(CommandSettings command);
+        Task GetPoolId(AgentSettings agentSettings, CommandSettings command);
 
         string GetFailedToFindPoolErrorString();
```

Comment:
manager does not throw with "GetFailedToFindPoolErrorString" it just prints the error and loop again